firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -1011,4 +1015,18 @@ public String toString() {
       }
     }
   }
+
+  static class DeserializeContext {
```

Comment:
It looks like this isn't used yet.  If you're planning to check this PR in in its current state, I think I'd suggest removing this for now and adding it in the PR that uses it?  That way:
1) we don't have dead code sitting around
2) you don't end up revamping this in the next PR once you actually use it (e.g. "documentId" should probably be a DocumentReference or DocumentKey now?)

Reply:
Deleted.